RecoJets/JetProducers/plugins/HFJetShowerShape.cc

```diff
@@ -0,0 +1,240 @@
+// -*- C++ -*-
+//
+// Package:    HFJetShowerShape/HFJetShowerShape
+// Class:      HFJetShowerShape
+//
+/**\class HFJetShowerShape HFJetShowerShape.cc HFJetShowerShape/HFJetShowerShape/plugins/HFJetShowerShape.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Laurent Thomas
+//         Created:  Tue, 25 Aug 2020 09:17:42 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/JetReco/interface/PFJet.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+
+#include "DataFormats/Math/interface/deltaPhi.h"
+#include "TMath.h"
+#include <iostream>
+
+//
+// class declaration
+//
+
+class HFJetShowerShape : public edm::stream::EDProducer<> {
+public:
+  explicit HFJetShowerShape(const edm::ParameterSet&);
+  ~HFJetShowerShape() override;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void produce(edm::Event&, const edm::EventSetup&) override;
+  template <typename T>
+  void putInEvent(const std::string&, const edm::Handle<edm::View<reco::Jet>>&, std::vector<T>*, edm::Event&) const;
+
+  edm::EDGetTokenT<edm::View<reco::Jet>> jets_token_;
+  edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_token_;
+
+  //Jet pt/eta thresholds
+  double jetPtThreshold_, jetEtaThreshold_;
+  //HF geometry
+  double hfTowerEtaWidth_, hfTowerPhiWidth_;
+  //Variables for PU subtraction
+  double vertexRecoEffcy_, offsetPerPU_, jetReferenceRadius_;
+  //Pt thresholds for showershape variable calculation
+  double stripPtThreshold_, widthPtThreshold_;
```

Comment:
```suggestion
  const edm::EDGetTokenT<edm::View<reco::Jet>> jets_token_;
  const edm::EDGetTokenT<std::vector<reco::Vertex>> vertices_token_;

  //Jet pt/eta thresholds
  const double jetPtThreshold_, jetEtaThreshold_;
  //HF geometry
  const double hfTowerEtaWidth_, hfTowerPhiWidth_;
  //Variables for PU subtraction
  const double vertexRecoEffcy_, offsetPerPU_, jetReferenceRadius_;
  //Pt thresholds for showershape variable calculation
  const double stripPtThreshold_, widthPtThreshold_;
```
all of this is known at configuration time and does not need to change